src/python/pants/backend/python/packaging/pyoxidizer/rules_integration_test.py

```diff
@@ -71,6 +71,46 @@ def test_end_to_end() -> None:
         assert bin_stdout == b"Hello world!\n"
 
 
+def test_run_binary() -> None:
+    sources = {
+        "hellotest/main.py": dedent(
+            """\
+            import sys
+
+            print('Hello world!')
+            sys.exit(42)
```

Comment:
Good call. Definitely do feel free to keep this line when merging into `test_end_to_end`. Note that `./pants package` should still be 0, but the `subprocess.run()` should be 42.

Reply:
I'll be merging this in the morning when I'm back at it.